### DISCUSSION TITLE
Live update inside table

### DISCUSSION THREAD

**Discussion by OskarKlintrot** [spectreconsole/spectre.console#274] [Q&A] [answered]
Is it possible to update a cell value inside a table once it's been rendered?

**Comment by rainersigwald** [comment]
I ran into this too in an experiment. I used private reflection to reach into `TableRow._items` but that does not seem like the intended design.

I just saw https://github.com/khalidabuhakmeh/StonksWatcher/pull/1 ([on Twitter](https://twitter.com/firstdrafthell/status/1416049650466017285)). @patriksvensson, is that "redefine the entire table then call `ctx.UpdateTarget(table)`" approach the "official" recommendation for this case?

**Reply by patriksvensson** [reply]
@rainersigwald Yes, that is the current official recommendation for this case.

**Comment by patriksvensson** [comment, marked as answer]
In version 0.42, there will be support for inserting rows and removing individual rows from a table, which might solve the original issue.

**Reply by pjsgsy** [reply]
Hooray! For info, I was trying to do this and came across this thread. It looks like there might be a way to do it from the initial post here, but I cannot quite figure it out! If possible and you could point me in the right direction, I'd appreciate it. Also, thanks for this project. It was just what I was looking for. Make me feel very nostalgic about my old dos days! :)

**Reply by pjsgsy** [reply]
OK, I think I figured it out and just stumbled across 'Live'.

**Comment by Rekkonnect** [comment]
I have a use case where I want to live update the progress of a process showing its time, and every time I have to reconstruct the entire table anew, which is pretty bad. It would be nice to only edit specific cells, which functionality is currently not present. Please either consider this for this discussion, or create a new issue to track it.
